ansible/playbooks/roles/grafana/tasks/upgrade/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+- name: Include new version of defaults
+  include_vars:
+    file: defaults/versions.yml
+    name: grafana_versions
+
+- name: Include upgrade defaults
+  include_vars:
+    file: defaults/upgrade.yml
+    name: upgrade_defaults
+
+- name: Populate service facts
```

Comment:
We need `ansible_facts.packages['grafana']` to get Grafana version so we need to use `package_facts` module. Hence there is no need to use `service_facts` since having grafana package present we will have `grafana.service` as well.